### ISSUE TITLE
Cycle gan g_loss and d_loss become nan after some epochs during training

### ISSUE THREAD

**Issue by Ayesha-Rafiq** [XHUJOY/CycleGAN-tensorflow#32]
Generator loss and Discriminator loss become nan after some epochs and cycle gan start to generate black images.

**Comment by Auth0rM0rgan** [comment]
Check your optimizer section, It may be related to set something wrong in optimizer.

**Comment by hala3** [comment]
i want to change the loss function by pixel wise loss , how i can do that please

**Comment by nagaswethar** [comment]
self.g_loss_a2b = self.criterionGAN(self.DB_fake, tf.ones_like(self.DB_fake)) \
        + self.L1_lambda * abs_criterion(self.real_A, self.fake_A_) \
        + self.L1_lambda * abs_criterion(self.real_B, self.fake_B_) \
        + self.Lg_lambda * gradloss_criterion(self.real_A, self.fake_B, self.weighted_seg_A) \
        + self.Lg_lambda * gradloss_criterion(self.real_B, self.fake_A, self.weighted_seg_B)
    self.g_loss_b2a = self.criterionGAN(self.DA_fake, tf.ones_like(self.DA_fake)) \
        + self.L1_lambda * abs_criterion(self.real_A, self.fake_A_) \
        + self.L1_lambda * abs_criterion(self.real_B, self.fake_B_) \
        + self.Lg_lambda * gradloss_criterion(self.real_A, self.fake_B, self.weighted_seg_A) \
        + self.Lg_lambda * gradloss_criterion(self.real_B, self.fake_A, self.weighted_seg_B)

can any one explain this loss function and what \ indicates in code
